website/static/js/contribRemover.js

```diff
@@ -0,0 +1,294 @@
+        }, self);
+
+        self.titlesToRemove = ko.computed(function() {
+            var titlesToRemove = [];
+            for (var key in self.nodesOriginal()) {
+                if (self.nodesOriginal().hasOwnProperty(key) && self.canRemoveNodes()[key]) {
+                    var node = self.nodesOriginal()[key];
+                    var contributors = node.contributors;
+                    for (var i = 0; i < contributors.length; i++) {
+                        if (contributors[i].id === self.contributorToRemove().id) {
+                            titlesToRemove.push(node.title);
+                            break;
+                        }
+                    }
+                }
+            }
+            return titlesToRemove;
+        });
+
+        self.titlesToKeep = ko.computed(function() {
+            var titlesToKeep = [];
+            for (var key in self.nodesOriginal()) {
+                if (self.nodesOriginal().hasOwnProperty(key) && !self.canRemoveNodes()[key]) {
+                    var node = self.nodesOriginal()[key];
+                    var contributors = node.contributors;
+                    for (var i = 0; i < contributors.length; i++) {
+                        if (contributors[i].id === self.contributorToRemove().id) {
+                            titlesToKeep.push(node.title);
+                            break;
+                        }
+                    }
+                }
+            }
+            return titlesToKeep;
+        });
+
+        self.nodeIDsToRemove = ko.computed(function() {
+            var nodeIDsToRemove = [];
+            for (var key in self.nodesOriginal()) {
+                if (self.nodesOriginal().hasOwnProperty(key) && self.canRemoveNodes()[key]) {
+                    var node = self.nodesOriginal()[key];
+                    var contributors = node.contributors;
+                    for (var i = 0; i < contributors.length; i++) {
+                        if (contributors[i].id === self.contributorToRemove().id) {
+                            nodeIDsToRemove.push(node.id);
+                            break;
+                        }
+                    }
+                }
+            }
+            return nodeIDsToRemove;
+        });
+
+        $.ajax({
+            url: window.contextVars.node.urls.api + 'tree',
+            type: 'GET',
+            dataType: 'json'
+        }).done(function(response) {
+            nodesOriginal = getNodesOriginal(response[0], nodesOriginal);
+            self.nodesOriginal(nodesOriginal);
+        }).fail(function(xhr, status, error) {
+            $osf.growl('Error', 'Unable to retrieve projects and components');
+            Raven.captureMessage('Unable to retrieve projects and components', {
+                url: self.nodeApiUrl, status: status, error: error
+            });
+        });
+    },
+    selectRemove: function() {
```

Comment:
I think I need it to build out the nodesOriginal list.  I could get a list but I know I am going to use the tree in add contributors later, and would prefer to use the same route until V2 is done.